HW31-CoreData/HW31App/HW31App/Services/CoreDataStorage.swift

```diff
@@ -0,0 +1,92 @@
+import UIKit
+import CoreData
+
+protocol BirthdayStorageProtocol {
+    func fetchBirthdayModels() -> [BirthdayInfo]
+    func getBirthday(by id: UUID) -> BirthdayInfo?
+    func deleteBirthday(by id: UUID)
+    func addBirthday(_ birthdayInfo: BirthdayInfo)
+    func editBirthday(_ birthdayInfo: BirthdayInfo)
+}
```

Comment:
Здесь все отличное, единственное что если хотим претендовать на мидлов-помидоров, то еще надо бы учесть, что могут быть ошибки. 
Например: 
```suggestion
enum MyBeautifulError: Error {
    case dataReadError 
}

protocol BirthdayStorageProtocol {
    func fetchBirthdayModels() -> Result<[BirthdayInfo], MyBeautifulError>
    func getBirthday(by id: UUID) -> Result<BirthdayInfo?, MyBeautifulError>
    func deleteBirthday(by id: UUID) ->  Result<Void, MyBeautifulError>
    func addBirthday(_ birthdayInfo: BirthdayInfo) -> Result<Void, MyBeautifulError>
    func editBirthday(_ birthdayInfo: BirthdayInfo) -> Result<Void, MyBeautifulError>
}
```